bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/types/XSTime.java

```diff
@@ -119,6 +119,9 @@ public static CalendarType parse_time(String str) {
 		if (dt == null)
 			return null;
 
+		// Adjust for 24:00 (xs:time only) by setting it back to 00:00 without rolling the date forward
+		dt.calendar().set(Calendar.DAY_OF_MONTH, 29);
+
```

Comment:
If we are parsing an `xs:dateTime`, then we want the date to roll forward because the date is relevant. `xs:time` differs in that it only represents a time, and that time always occurs within the bounds of a fixed day (in this implementation, it's Nov. 29, 1983). The need for a special case arises because we use the same parsing code for both `xs:dateTime` and `xs:time`.